ctmap: Fix conntrack map filtering

tgraf opened the pull request:
The type was incorrect causing all filters to be ignored. Add a warning
statement to catch future mismatches.

Fixes: eacec0bffcf ("pkg/maps: create CtKeyGlobal structures")

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/8182)
<!-- Reviewable:end -->

aanm reviewed (changes requested):
@tgraf what about *tuple.TupleKey4 a couple lines bellow?

tgraf committed:
ctmap: Fix conntrack map filtering

The type was incorrect causing all filters to be ignored. Add a warning
statement to catch future mismatches.

Fixes: eacec0bffcf ("pkg/maps: create CtKeyGlobal structures")

Signed-off-by: Thomas Graf <thomas@cilium.io>
